docs/content/basic/app-assistant.md

```diff
@@ -0,0 +1,156 @@
+---
+title: Agents & Assistants
+lang: en
+slug: /concepts/assistant
+---
+
+:::info[This feature requires a paid plan]
+If you don't have a paid workspace for development, you can join the [Developer Program](https://api.slack.com/developer-program) and provision a sandbox with access to all Slack features for free.
+:::
+
+Agents and assistants comprise a new messaging experience for Slack. If you're unfamiliar with using agents and assistants within Slack, you'll want to read the [API documentation on the subject](https://api.slack.com/docs/apps/ai). Then, come back here to implement them with Bolt!
+
+## Configuring your app to support assistants 
+
+1. Within [App Settings](https://api.slack.com/apps), enable the **Agents & Assistants** feature.
+
+2. Within the App Settings **OAuth & Permissions** page, add the following scopes: 
+  * [`assistant:write`](https://api.slack.com/scopes/assistant:write)
+  * [`chat:write`](https://api.slack.com/scopes/chat:write)
+  * [`im:history`](https://api.slack.com/scopes/im:history)
+
+3. Within the App Settings **Event Subscriptions** page, subscribe to the following events: 
+  * [`assistant_thread_started`](https://api.slack.com/events/assistant_thread_started)
+  * [`assistant_thread_context_changed`](https://api.slack.com/events/assistant_thread_context_changed)
+  * [`message.im`](https://api.slack.com/events/message.im)
+
+:::info
+You _could_ implement your own assistants by [listening](/concepts/event-listening) for the `assistant_thread_started`, `assistant_thread_context_changed`, and `message.im` events. That being said, using the `Assistant` class will streamline the process. And we already wrote this nice guide for you!
+:::
+
+## The `Assistant` class instance
+
+```ts
+const assistant = new Assistant({
+  threadContextStore: {
```

Comment:
We can make the minimam required one as simple as possible. You can go with your own comment though!

```suggestion
  // This is optional; if you don't have it, the built-in default implementation works welll for you
  threadContextStore: {
```

Reply:
The default one is purely in-memory, right? If so, it will not work well when used with the `AwsLambdaReceiver`, since each event invocation spins up a fresh Bolt instance, meaning the default thread context store will be empty each time. If so, we might need a callout here to instruct users of the `AwsLambdaReceiver` to not use the default one.

Reply:
@filmaj the default one uses the bot's first reply's message metadata to remember the latest context data, so you can safely use it for any situation.